### PR TITLE
[4.0] crowbar_framework:fix filtered nodes in apply_role

### PR DESCRIPTION
**Pull request by Itxaka**
As a side effect of introducing the experimental options for unchanged
and unready nodes into apply_role, there could be some cases in which
even if the nodes where removed from the list to be deployed we still
deployed them. For that issue, a check is done before adding the nodes
into the batches to check that indeed those nodes are going to be
deployed.

There was also a wrong comparision being done of the old_nodes vs
new_nodes in which we used the filtered list of nodes, but we want to
use the original node list without unfiltered to make sure that the
deployments didnt change. For this,the unfiltered list of nodes in the
new deployment is regenerated and used for the role comparision.

(cherry picked from commit adaa2b548543faa3c88c7262415bde369c1f7650)
Backport-of: https://github.com/crowbar/crowbar-core/pull/1401

### CODE REVIEW

**Comment by Itxaka** [comment]
[Skip batches when applying nova barclamp](https://trello.com/c/rYkjyvHO/161-skip-batches-when-applying-nova-barclamp)

**Comment by Itxaka** [comment]
@vuntz cherry-picked latest commit with fixes from master

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -1188,6 +1202,9 @@ def apply_role(role, inst, in_queue, bootstrap = false)
               next
             end
 
+            # skip adding nodes to the batch unless they are really in the list to be deployed
+            next unless new_elements[role_name].include?(node_name)
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: Not a clean cherry-pick :-)

Comment (Itxaka) [reply crowbar_framework/app/models/service_object.rb]: weeeeird, thats what I get for doing things quick without double checking :(

Comment (Itxaka) [reply crowbar_framework/app/models/service_object.rb]: fixed!

**Pull request merged**